packages/xchain-evm/src/cache.ts

```
@@ -0,0 +1,62 @@
+ */
+async function getContractCacheKey(address: string, provider: Provider): Promise<string> {
+  try {
+    // Get network information from provider to create unique key
+    const network = await provider.getNetwork()
+    const chainId = network.chainId.toString()
+    const networkName = network.name || 'unknown'
+    return `${networkName}_${chainId}_${address.toLowerCase()}`
+  } catch {
+    // Fallback to a provider-specific key if network info unavailable
+    // Use provider instance as unique identifier
+    const providerIdentity = provider as any
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)